src/fidesops/main.py

```diff
@@ -40,6 +42,75 @@
         allow_headers=["*"],
     )
 
+
+@app.middleware("http")
+async def dispatch_log_request(request: Request, call_next: Callable) -> Response:
+    fides_source: Optional[str] = request.headers.get("X-Fides-Source")
+    now: datetime = datetime.now(tz=timezone.utc)
+    endpoint = f"{request.method}: {request.url.path}"
+
+    try:
+        response = await call_next(request)
+        # HTTPExceptions are considered a handled err by default so are not thrown here.
+        # Accepted workaround is to inspect status code of response.
+        # More context- https://github.com/tiangolo/fastapi/issues/1840
+        try:
+            if response.status_code >= 400:
+                response.background = BackgroundTask(
+                    prepare_and_log_request,
+                    endpoint,
+                    response.status_code,
+                    now,
+                    fides_source,
+                    "HTTPException",
+                )
+            else:
+                response.background = BackgroundTask(
+                    prepare_and_log_request,
+                    endpoint,
+                    response.status_code,
+                    now,
+                    fides_source,
+                    None,
+                )
+        except Exception as exc:
+            # always continue if something went wrong with analytics
+            logger.warning(
+                f"Analytics event for endpoint {request.url} failed to send: {exc}"
+            )
+            return response
```

Comment:
Will we actually catch anything here in practice, since it looks like `send_analytics_event` also has a try/except?  It would be nice to get rid of this doubly-nested try/except.

Reply:
This `Exception` may catch issues with `BackgroundTask`. It was helpful when debugging async issues with tasks, but I guess it's not needed now that it's properly working. I can remove!